src/Functions/max2.cpp

```diff
@@ -0,0 +1,22 @@
+#include <Functions/FunctionMathBinaryFloat64.h>
+#include <Functions/FunctionFactory.h>
+
+template <typename T>
+T max(T a, T b)
```

Comment:
No, if it can I would of course use it. Exception message: `Value of type '<overloaded function type>' is not implicitly convertible to 'DB::Float64 (*)(DB::Float64, DB::Float64)' (aka 'double (*)(double, double)')`.